File: lol/src/main/kotlin/com/your/lol/dto/riot/InfoDto.java
package com.your.lol.dto.riot;

import java.util.List;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class InfoDto {
    private List<ParticipantDto> participants;
    private List<TeamDto> teams;

    public ParticipantTeamDto filterAndCreateCompositeDto(String summonerName) {
        ParticipantDto findParticipant = participants.stream()
                .filter(participant -> participant.equalsId(summonerName))
                .findAny()
                .orElseThrow(() -> new IllegalArgumentException("챔피언을 찾을 수 없습니다."));
        TeamDto findTeam = teams.stream()
                .filter(team -> team.equalsId(findParticipant.getTeamId()))
                .findAny()
                .orElseThrow(() -> new IllegalArgumentException("팀을 찾을 수 없습니다."));
        return new ParticipantTeamDto(findParticipant, findTeam.isWin());
    }
}

File: ggtics-recruitement-lol-score/src/main/kotlin/com/your/lol/dto/riot/InfoDto.kt
package com.your.lol.dto.riot

import java.util.Collections.emptyList

class InfoDto(
        val participants: List<ParticipantDto>,
        val teams: List<TeamDto>) {

    private constructor() : this(emptyList(), emptyList())

    fun filterAndCreateCompositeDto(summonerName: String): ParticipantTeamDto {
        val findParticipant = findParticipant(summonerName)
        val findTeam = getFindTeam(findParticipant)
        return ParticipantTeamDto(findParticipant, findTeam.win)
    }

    private fun findParticipant(summonerName: String): ParticipantDto = participants.stream()
            .filter { it.equalsId(summonerName) }
            .findAny()
            .orElseThrow { IllegalArgumentException("챔피언을 찾을 수 없습니다.") }

    private fun getFindTeam(findParticipant: ParticipantDto): TeamDto =
            teams.stream()
                    .filter { it.equalsId(findParticipant.teamId) }
                    .findAny()
                    .orElseThrow { IllegalArgumentException("팀을 찾을 수 없습니다.") }
}